packages/front-end/components/SchemaBrowser/SqlExplorerModal.tsx

```diff
@@ -304,6 +307,18 @@ export default function SqlExplorerModal({
     }
   };
 
+  const getTruncatedTitle = (
+    title: string,
+    totalVisualizationCount: number
+  ): string => {
+    // Only truncate if there are 4 or more visualizations
+    if (totalVisualizationCount < 4) return title;
+
+    const maxLength = 20;
+    if (title.length <= maxLength) return title;
+    return title.substring(0, maxLength) + "...";
```

Comment:
very nit: should it be `maxLength - 3` to account for the added `...`?

Otherwise the text will be longer in the cases of 21 or 22 characters.

Reply:
The 20 char limit isn't anything special, just an arbitrary limit Kari and I landed on, so it's safe, but I think this is a good callout.